content/blog-posts/2024-06-13-ebpf-licensing/index.md

```diff
@@ -0,0 +1,50 @@
+---
+path: '/blog/ebpf-licensing-guide'
+date: '2024-06-13T10:00:00.000Z'
+title: 'A Practical Guide to eBPF Licensing: Or How I Learned to Stop Worrying and Love the GPL'
+description: "Learn how to license eBPF programs and their userspace components with the legal considerations along the way"
+ogImage: modular-vs-aggregate.png
+categories:
+  - Community
+---
+
+
+_Disclaimer: We are not lawyers and this is not legal advice. Please contact a lawyer for legal advice._
+
+Authors: Bill Mulligan and Liz Rice
+
+While there is a wide appreciation of the power of eBPF to [fundamentally change the platforms](https://www.infoq.com/articles/ebpf-cloud-native-platforms/) we run our software on, there are some misconceptions in the ecosystem about when eBPF programs need to be licensed under the GPL and how that impacts licensing compliance. Seeing GPL licensed code may give some legal teams pause because eBPF is unfamiliar to them - but it shouldn’t. This blog provides context around why GPL licensed eBPF code exists and lays out why if you are already using Linux in your project, product, or company then using eBPF should not cause any additional concerns. If you are a contributor to a project already using eBPF or looking to add it, or you're unfamiliar with or concerned about eBPF licensing, this blog will walk you through some practical strategies for eBPF licensing and hopefully you’ll learn to stop worrying and love the GPL.
+
+## All Meaningful eBPF Programs Are GPL Licensed
```

Comment:
```suggestion
## All Meaningful Linux eBPF Programs Are GPL Licensed
```
This doesn't apply to other platforms like ebpf-for-windows, where meaningful linux eBPF programs need not be GPL licensed.

Reply:
I'm adding kernel here too